### ISSUE TITLE
Compilation problem with macro expansion

### ISSUE THREAD

**Issue by JeanLucPons** [tango-controls/pytango#196]
Hello,

I'm trying to compile pytango with gcc 6.3.0 on debian9 and it fails during a macro expansion:

```
ext/device_pipe.cpp:137:13: note: in expansion of macro ‘TANGO_DO_ON_DEVICE_DATA_TYPE_ID’
             TANGO_DO_ON_DEVICE_DATA_TYPE_ID(elt_type,
             ^
ext/device_pipe.cpp: In function ‘boost::python::api::object PyTango::DevicePipe::__extract_item(T&, size_t, PyTango::ExtractAs)’:
ext/tgutils.h:234:73: error: ‘DOIT_ARRAY’ was not declared in this scope
 ine __TANGO_DVBA __TANGO_DEPEND_ON_TYPE_AUX_ID(DEVVAR_BOOLEANARRAY, DOIT_ARRAY)
```
May be I missed something....
Thanks for help.
Jean-Luc PONS

**Comment by JeanLucPons** [comment]
I manage to make it work by commenting out few lines of `ext/tgutils.h`
I compiled with tango 9.3.1.

```c++
//#if TANGO_VERSION_NB > 90205
//#define __TANGO_DVBA __TANGO_DEPEND_ON_TYPE_AUX_ID(DEVVAR_BOOLEANARRAY, DOIT_ARRAY)
//#else
#define __TANGO_DVBA
//#endif
```

**Comment by vxgmichel** [comment]
Hi @JeanLucPons, thanks for the report!

@grm84 , you seem to be the author of this code:

https://github.com/tango-controls/pytango/commit/0911cbf1986884a707a45cb194fb25db1d0cbaee#diff-963e048d92def45561ed029398f47ea7R233

Any idea what it is about?

**Comment by grm84** [comment]
Hi Vincent,


This piece of code was introduced to add booleanarrays which did not exist prior to Tango release 9.2.5.

It was to try and keep backward compatibility.


I did see Jean-Luc's report but I've not had chance to look at it and I also don't have that modern compiler.

Can I access it with Conda?


Regards, Geoff.

________________________________

**Comment by vxgmichel** [comment]
> Can I access it with Conda?

Yes, we're currently using g++ 7.2.0 to build our packages:

    $ conda install gxx_linux-64

But what makes you think this issue is related to the compiler? 

It seems to me that the condition `if TANGO_VERSION_NB > 90205` has never been met before and that the following line is causing the problem.

**Comment by vxgmichel** [comment]
Fixed in PR #197.
